### DISCUSSION TITLE
visidata on windows

### DISCUSSION THREAD

**Discussion by vanderdoes** [saulpw/visidata#2498] [Q&A]
Hi,

I believe I have use cases for trying visidata.  I tried installing wsl but i believe corporate policies prevent it from being installed.  I am a bit baffled on MSYS2 approach.  Can use this approach and have it not mess up my python environment?  I have looked at web site but am missing how to install/use this approach.  Any pointers?

Thanks!
Jeff

**Comment by cool-RR** [comment]
I couldn't get VisiData to work well on Windows with the MSYS2 shell without WSL. I doubt it's possible or going to be possible anytime soon.

To clarify, it might be possible with some kind of VM or remote computer.

**Comment by ajkerrigan** [comment]
You _should_ be able to install it from a native cmd or powershell prompt as long as you've got Python installed. Worth trying:

```console
python -m venv visidata

.\visidata\Scripts\activate.bat  # For cmd
.\visidata\Scripts\activate.ps1  # For powershell

pip install visidata
visidata
```

Seems to work on the native powershell or cmd on my (now pretty old and dusty) Windows 10 machine at least, maybe for you too? 🤞

**Reply by cool-RR** [reply]
It's not only about whether VisiData starts or not. When I ran it on Windows, it did start. But:

* Not all the hotkeys worked properly: https://github.com/saulpw/visidata/issues/1630 (There are more hotkeys that don't work well that are not in this bug)
* Some of the colors were off: https://github.com/saulpw/visidata/issues/1746
* It would sometimes segfault, especially when I changed the terminal size.

**Reply by ajkerrigan** [reply]
Whoops, misunderstood the question. Gotcha.
